docs/DeveloperGuide.md

```diff
@@ -849,3 +849,25 @@ Note that to try out the below example commands you will need to reset the data
+Implementing the clickable detailed view page was another challenge as we had to create a new page and our team generally found difficulty in dealing with the UI. Implementing headless testing for UI components so that code coverage can be satisfied for GitHub CI/CD was a challenge. It required a whole new library dependency and framework (TestFX) for testing UI components which was not easily referenced from the existing codebase. If an internship was selected via the detailed view, and it was simultaneously edited in the the main window, the internship would disappear from the detailed view window. After prolonged debugging, the problem was isolated to the commands that edit the 'state' of the Internship. The issue was resolved by fixing the equality checks used by the detailed view page.
+#### Effort Required
+As a team, we have put a lot of effort into this project. Firstly through the process of examining and understanding the codebase and coming up with useful features that could improve user experience. We had to streamline our ideas and come up with a feasible plan to manage the workload.
+Following this planning out user requirements, user stories, and use cases was something new to us and we spent a fair bit of time on this. As for the actual implementation, while we are not new to programming, a brownfield collaborative project is unfamiliar territory. In the first few weeks, we made countless errors when it came to setting up build tools and adhering to the general workflow.
+Finally, the documentation required was extensive and required careful examination. This is something we had to sit down and go through together as a team multiple times.
+#### Achievements
+Despite the challenges encountered, the team achieved several milestones that significantly strengthened the application`s functionality and user experience. Some of these key achievements include:
+Successful integration of optional field capabilities for an internship entry, offering users more flexibility in adding and editing internship entries.
+Redesigning command formats to cater to our target users and improve user experience.
+Implementation of find and sorting functionalities, encouraging and enabling users to use the application to navigate and manage their internship entries efficiently.
+Introduction of a remark feature, ensuring that the application facilitates the inclusion of additional information.
+Introduction of a task feature, allowing users to add tasks relating to an internship entry and add suitable deadlines.
+#### Effort Saved Through Reuse:
+While we did not have a lot of code reuse or use of external dependencies, we used the TestFx (an extension of JavaFx) library to implement headless testing to write automated test cases for our UI components. This was done to ensure that we don’t sacrifice on code coverage of our tests.
```

Comment:
"... sacrifice on code coverage of our tests for the new Ui components, to ensure they work perfectly well".

^ Expand the sentence as such